### PR TITLE
Fix Executor @handler validation for postponed annotations

### PR DESCRIPTION
**Pull request by moonbox3**
## Summary
Fixes Executor `@handler` signature validation when `from __future__ import annotations` is enabled.

## Root cause
`_validate_handler_signature` validated `ctx_param.annotation` directly from `inspect.signature`, which can be a `str` under postponed annotations. This caused valid `WorkflowContext[...]` annotations to be rejected.

## Changes
- Resolve type hints via `typing.get_type_hints(..., include_extras=True)` in `_validate_handler_signature`.
- Validate `WorkflowContext[...]` using the resolved ctx annotation.
- Provide a clearer error message when annotations cannot be resolved.

## Tests
- Added regression test ensuring `WorkflowContext[T, U]` is accepted and types are inferred under postponed annotations.
- Added edge-case test asserting a clear error when forward references cannot be resolved.

## Verification note
A required verification stage is currently failing because it invokes an unconfigured Poe task (`pre-commit-check`). This patch does not modify CI/infrastructure/config per constraints; it targets the Executor behavior and adds tests.

### CODE REVIEW

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 1]
```diff
@@ -717,25 +717,58 @@ def _validate_handler_signature(
...
+        import sys
+        import typing
+
+        module = sys.modules.get(getattr(func, "__module__", ""))
+        module_ns = getattr(module, "__dict__", None) if module is not None else None
+
+        globalns: dict[str, Any] = module_ns or getattr(func, "__globals__", {})
+        localns: dict[str, Any] = module_ns or globalns
+
+        type_hints = typing.get_type_hints(func, globalns=globalns, localns=localns, include_extras=True)
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:734]: Catching broad `Exception` around `typing.get_type_hints` and forcing `type_hints = {}` can mask real bugs and later route into misleading fallback errors; restrict to expected resolution failures (e.g., NameError/TypeError) and re-raise unexpected exceptions.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 2]
```diff
@@ -717,25 +717,58 @@ def _validate_handler_signature(
...
+        type_hints = {}
+
     # Check message parameter has type annotation (unless skipped)
     message_param = params[1]
     if not skip_message_annotation and message_param.annotation == inspect.Parameter.empty:
         raise ValueError(f"Handler {func.__name__} must have a type annotation for the message parameter")
 
+    resolved_message_annotation = type_hints.get(message_param.name, message_param.annotation)
+
     # Validate ctx parameter is WorkflowContext and extract type args
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:745]: The emptiness check uses `ctx_param.annotation` (raw) but later validates `resolved_ctx_annotation`; with postponed annotations the raw value may be a string/ForwardRef while the resolved annotation is valid—use the resolved annotation consistently to avoid incorrect rejection/branching.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 3]
```diff
@@ -717,25 +717,58 @@ def _validate_handler_signature(
...
+        localns: dict[str, Any] = module_ns or globalns
+
+        type_hints = typing.get_type_hints(func, globalns=globalns, localns=localns, include_extras=True)
+    except Exception:
+        type_hints = {}
+
     # Check message parameter has type annotation (unless skipped)
     message_param = params[1]
     if not skip_message_annotation and message_param.annotation == inspect.Parameter.empty:
         raise ValueError(f"Handler {func.__name__} must have a type annotation for the message parameter")
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:742]: If `get_type_hints()` fails under postponed annotations, `resolved_message_annotation` may remain a `str` and `message_type` can become that string; detect unresolved string annotations and raise a clear error (or ensure resolution) rather than silently proceeding.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 4]
```diff
@@ -717,25 +717,58 @@ def _validate_handler_signature(
...
-    message_type = message_param.annotation if message_param.annotation != inspect.Parameter.empty else None
-    ctx_annotation = ctx_param.annotation
+        try:
+            output_types, workflow_output_types = validate_workflow_context_annotation(
+                resolved_ctx_annotation, f"parameter '{ctx_param.name}'", "Handler"
+            )
+        except ValueError as exc:
+            # If we couldn't resolve annotations, provide a clearer error than the generic
+            # WorkflowContext validation failure.
+            if isinstance(resolved_ctx_annotation, str) or isinstance(ctx_param.annotation, str) or not type_hints:
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:762]: This branch raises “type annotation could not be resolved” when `not type_hints`, which conflates resolution failure with real validation errors; distinguish resolution failure from `validate_workflow_context_annotation` failures so error messages remain accurate.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    pass
+
+
+class _FutureExecutor(Executor):
+    @handler
+    async def example(self, input: str, ctx: WorkflowContext[_TypeA, _TypeB]) -> None:
+        return None
+
+
+def test_executor_handler_future_annotations_resolves_workflow_context_generics() -> None:
```
Comment (moonbox3) [inline python/packages/core/tests/workflow/test_executor_future_annotations.py:25]: To prove the regression, assert postponed annotations are actually strings on the signature (e.g., `inspect.signature(...).parameters['ctx'].annotation` is `str`/contains `WorkflowContext[...]`), otherwise the test may pass even without the fix.

**Review by moonbox3 — COMMENTED** [review]
## Automated PR Review Round 1: CHANGES_REQUESTED

Issue: #1
The executor change to resolve postponed annotations via `typing.get_type_hints` is on the right track, but the current implementation can silently accept unresolved annotations, produce misleading errors due to over-broad exception handling, and inconsistently validates raw vs resolved annotations. Additionally, the PR introduces new GitHub Actions workflows that execute cloned code with privileged secrets, which is a major security risk and unrelated to the executor fix. Tests added are helpful but don’t sufficiently prove the regression surface or cover key edge cases introduced by the new resolution logic.

Blocking issues (5):
- Remove or split out the newly added GitHub Actions workflows that clone and execute repository code while providing high-privilege secrets (RCE/secret-exfiltration risk); this is unrelated to the executor fix and is not safe to merge as-is.
- `_validate_handler_signature` inconsistently checks emptiness using the raw `ctx_param.annotation` but validates `resolved_ctx_annotation`; under postponed annotations this can incorrectly reject valid handlers or raise in the wrong branch—use the resolved annotation consistently for the emptiness/validation decisions.
- `typing.get_type_hints` exceptions are caught too broadly and then converted into a generic “could not be resolved” failure path, which can mask genuine validation errors and produce misleading error messages; narrow the exceptions and distinguish resolution failure from validation failure.
- When `get_type_hints` fails under postponed annotations, message annotations may remain strings and `message_type` can become a `str`, silently weakening validation and likely breaking downstream type logic; detect and fail clearly (or ensure message annotations are resolved) rather than proceeding with a string.
- +1 more

Suggestions (3):
- Limit `get_type_hints` exception handling to expected resolution errors (e.g., `NameError`, `TypeError`, possibly `AttributeError`) and re-raise unexpected exceptions to avoid hiding bugs.
- Consider calling `typing.get_type_hints(..., include_extras=True)` if the framework relies on `Annotated[...]`/extras, and ensure the namespace used supports resolving method-local/class-scoped types.
- Improve error reporting by preserving the underlying resolution error context (without leaking secrets) so failures are diagnosable and not uniformly reported as “could not be resolved.”

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 1 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:734]: Automated status: queued for remediation in this pass.

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 2 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:745]: Automated status: queued for remediation in this pass.

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 3 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:742]: Automated status: queued for remediation in this pass.

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 4 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:762]: Automated status: queued for remediation in this pass.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py (hunk 5 above)
Comment (moonbox3) [reply python/packages/core/tests/workflow/test_executor_future_annotations.py:25]: Automated status: queued for remediation in this pass.
